### PR TITLE
Fields Edit_Table Redo v1

### PR DESCRIPTION
**Pull request by kcarcia**
Addresses #2002, #1623, #1966.

This is only the first iteration of edit_fields, therefore the buttons to add fields are at the top. That's because I haven't addressed field ordering yet, and fields are still being added through form submission. The page refreshes every time a user adds a field, and I could load the page at the bottom, but then you don't see any success or error messages. That's why, for now, the buttons are at the top.

Moving the buttons to the bottom and reordering fields will come in my second iteration of this (which you can find here #2009)

@jdalphon I do have a concern regarding a change I made to the projects controllers. The concern comes up in one of the tests. I noted where my concern is in my code.

### CODE REVIEW

File: test/functional/projects_controller_test.rb [hunk 1]
```diff
@@ -241,8 +241,7 @@ class ProjectsControllerTest < ActionController::TestCase
                       '21_unit' => parameters['21_unit'], '22_name' => parameters['22_name'], '22_unit' => parameters['22_unit'],
                       "#{new_lat_field_id}_name" => @project.fields[num_fields - 2].name, "#{new_lat_field_id}_unit" => @project.fields[num_fields - 2].unit,
                       "#{new_long_field_id}_name" => @project.fields[num_fields - 1].name, "#{new_long_field_id}_unit" => @project.fields[num_fields - 1].unit,
-                      '20_restrictions' => 'a,b,c',
-                      '25_name' => 'Location of Foodz', '25_unit' => '', new_field: '' }, user_id: @kate.id
+                      '20_restrictions' => 'a,b,c' }, user_id: @kate.id
```
Comment (kcarcia) [inline test/functional/projects_controller_test.rb]: '25_name' => 'Location of Foodz', '25_unit' => '', new_field: '' "

This line was making the test fail. I don't understand why it's there. This test is creating that field and that line does nothing to change the field. Any thoughts?

File: app/controllers/projects_controller.rb [hunk 2]
```diff
@@ -297,20 +299,23 @@ def save_fields
         redirect_to "/projects/#{@project.id}/edit_fields"
         return
       end
-    elsif field_type != ''
-      next_name = Field.get_next_name(@project,
-                                      get_field_type(params[:new_field]))
-      field = Field.new(project_id: @project.id,
-                        field_type: get_field_type(field_type), name: next_name)
+    elsif !field_type.nil?
+      while i < ((params[:num_fields].to_i > params[:text_fields].to_i) ? params[:num_fields].to_i : params[:text_fields].to_i)
```
Comment (kcarcia) [inline app/controllers/projects_controller.rb]: Hoping implementing this won't change anything API related.

**Comment by stowellm** [comment]
@kcarcia - I pulled and tested this: looks great!  As you already hinted at, it still refreshes when adding fields so it'll be awesome to see that not happen on the next iteration.

The one thing I'm not so certain about is the "Field added" pop-up when you add a field.  The first time you add a field, it shows up - great, you know you're doing it right.  The next time you add a field - assuming you haven't X'ed out the pop-up - there's now no feedback.  "Field added" just stays up there.  Possible solutions could be a timer before dismissing it, or perhaps putting the names of the fields added so that at least the text changes every time you add a field.

**Comment by kcarcia** [comment]
@stowellm Great feedback, Mike. Both solutions are easy to implement. I will fix this tomorrow.

**Pull request merged**